Makefile

```diff
@@ -669,6 +669,14 @@ test-capd-junit: $(SETUP_ENVTEST) $(GOTESTSUM) ## Run unit and integration tests
 kind-cluster: ## Create a new kind cluster designed for development with Tilt
 	hack/kind-install-for-capd.sh
 
+.PHONY: kind-reset
+kind-reset: ## Destroys the "capi-test" kind cluster.
+	kind delete cluster --name=capi-test || true
```

Comment:
I dont think we can always assume that the kind cluster's name will be `capi-test` given that `make kind-cluster` can create clusters with other names. We can either enforce the cluster name as @killianmuldoon suggested or have the `kind-reset` target also support the `CAPI_KIND_CLUSTER_NAME` like the `kind-cluster` target does.

I dont have strong opinions on either options.

Reply:
I would prefer to use the same logic here as we have in the script. Which means use the env var `CAPI_KIND_CLUSTER_NAME` if it is set and fallback to `capi-test` if the env var is not set